packages/functional-tests/tests/resetPassword/resetPasswordWithCode/oauthResetPassword.spec.ts

```diff
@@ -0,0 +1,197 @@
+import { ResetPasswordReactPage } from '../../../pages/resetPasswordReact';
+import { SigninReactPage } from '../../../pages/signinReact';
+
+test.describe('severity-1 #smoke', () => {
+  test.describe('oauth reset password', () => {
+    test.beforeEach(async ({ pages: { configPage } }) => {
+      const config = await configPage.getConfig();
+      test.skip(
+        config.featureFlags.resetPasswordWithCode !== true,
+        'see FXA-9728, remove conditional skip when feature flag removed'
+      );
+    });
+
+    test('reset password', async ({
+      target,
+      page,
+      pages: { relier, resetPasswordReact, signinReact },
+      testAccountTracker,
+    }) => {
+      const credentials = await testAccountTracker.signUp();
+      const newPassword = testAccountTracker.generatePassword();
+
+      await relier.goto();
+      await relier.clickEmailFirst();
+
+      await beginPasswordReset(
+        resetPasswordReact,
+        signinReact,
+        credentials.email
+      );
+
+      const code = await target.emailClient.getResetPasswordCode(
+        credentials.email
+      );
+
+      await resetPasswordReact.fillOutResetPasswordCodeForm(code);
+      await resetPasswordReact.fillOutNewPasswordForm(newPassword);
+
+      await expect(page).toHaveURL(/reset_password_verified/);
+      await expect(
+        resetPasswordReact.passwordResetConfirmationHeading
+      ).toBeVisible();
+
+      // TODO in FXA-9561 - Verify that the service name is displayed in the "Continue to ${serviceName}" button
+      // This functionality is not yet implemented in the reset password flow
+
+      // TODO in FXA-9561 - Remove this temporary test of sign in with new password
+      // user should be signed in and able to navigate to the relying party
+
+      await relier.goto();
+      await relier.clickEmailFirst();
+      await signinReact.fillOutEmailFirstForm(credentials.email);
+      // old password fails
+      await signinReact.fillOutPasswordForm(credentials.password);
+      await expect(page.getByText('Incorrect password')).toBeVisible();
+      // new passwowrd works
+      await signinReact.fillOutPasswordForm(newPassword);
+      await expect(page).toHaveURL(target.relierUrl);
+      expect(await relier.isLoggedIn()).toBe(true);
+
+      // update password for cleanup function
+      credentials.password = newPassword;
+    });
+
+    test('reset password with PKCE', async ({
+      target,
+      page,
+      pages: { relier, resetPasswordReact, signinReact },
+      testAccountTracker,
+    }) => {
+      const credentials = await testAccountTracker.signUp();
+      const newPassword = testAccountTracker.generatePassword();
+
+      // TODO in FXA-8006 - Update this test to use the 123Done PKCE button once fixed
+      // For now navigate directly to the link.
+      await page.goto(
+        `http://localhost:3030/authorization?` +
+          `&access_type=offline` +
```

Comment:
tiny nit: this will generate a search query like `/authorization?&access_type=` (with `?&`).
Possibly not worth creating a new `URL()` instance and setting `query` param if an object syntax is easier to read.

```js
const uri = new URL("/authorization", "http://localhost:3030");
uri.search = new URLSearchParams({
  beep: 'boop',
  foo: 'bar',
});
page.goto(uri.toString());
```

Reply:
This is meant to be temporary since we have a ticket filed to fix the 123Done button - I'll just remove the stray `&` on the first param for now.